### PR TITLE
fix: Remove findDOMNode

### PR DESCRIPTION
**Pull request by mjpsyapse**
This change removes the deprecated findDOMNode function.

### CODE REVIEW

**Comment by mjpsyapse** [comment]
tl;dr Don't merge this.

I have been exploring the entire antd / react-component ecosystem, and findDOMNode is used heavily, which makes this migration require a lot of untangling, if not rewriting.  For example, `rc-animate` should be replaced with `react-transition-group`.  In general, I'd say that react-components and antd try to wrap existing libraries that do DOM manipulation with react (like `rc-align`).  It will be a large shift to refactor out all this history.

We need to figure out a strategy for moving forward or else `antd` will be left behind as `react` adds new features.  i.e. no suspense, no concurrent react, and whatever else comes next.

**Comment by zombieJ** [comment]
This can not get real dom node:
```jsx
class MyComponent {
  render() {
    return (<div {...this.props} />);
  }
}
```

`findDOMNode` will get instance instead and makes `getBoundingClientRect` failed.

In low level api, `findDOMNode` seems still necessary since user may not pass real dom ref out.

**Comment by mjpsyapse** [comment]
@zombieJ Yes, I agree that this is a breaking change.  And I agree that `findDOMNode` is useful, but if it's deprecated, there should be a plan to remove it.  Right?

**Comment by zombieJ** [comment]
Here are some info by React doc: https://reactjs.org/docs/forwarding-refs.html#forwarding-refs-in-higher-order-components

User should always pass dom with `forwardRef` and then `findDomNode` can be removed. But it also makes that if instance do provides some functions breaks. Or may require user use render props return node & ref.
Still thinking of it since it's a hard decision.

**Comment by mjpsyapse** [comment]
Closing this since it will never get merged.
